revscoring/utilities/fetch_idioms.py

```diff
@@ -55,15 +56,49 @@ def fetch(output):
         continuation=True)
 
     idioms = []
+    first_term = ''
+    other_terms = []
     for doc in results:
         for page_doc in doc['query']['categorymembers']:
             # Some values take the form:
             # Category:English <variable>
             # For example: Category:English rhetorical questions
             # They are not idioms and should be filtered
             idiom = page_doc['title']
-            if 'Category:English' not in idiom:
-                idioms.append(idiom)
+        if re.match('Category:English|Citation:|Appendix:', idiom):
+            continue
+        # Remove one word idioms
+        if re.match(r'^[\w-]+$', idiom):
+            continue
+        # Remove two-worded idioms
+        if re.match(r'^[\w-]+ [\w-]+$', idiom):
+            continue
+        # Remove similes
+        if 'as a' in idiom:
+            continue
+        terms = idiom.split()

```

Comment:
It seems like this section should be another function.  like:

```
def regexify(idioms):
  ...
```

It would return a generator that produces the regexes you'd like to output.  This will be handy because it'll make testing a lot easier.